sources/editor/Stride.Core.Assets.Editor/ViewModel/AssetCollectionViewModel.cs

```diff
@@ -537,31 +535,99 @@ private string ComputeNamespace(DirectoryBaseViewModel directory)
+                        {
+                            message = Tr._p("Message", "The selected directory is not a subdirectory of the resources folder!");
+                            await Dialogs.MessageBox(message, MessageBoxButton.OK, MessageBoxImage.Error);
                         }
                     }
 
+                } while (!done);
+            }
+            return finalPath;
+        }
+
+        private async Task<List<AssetViewModel>> InvokeAddAssetTemplate(LoggerResult logger, string name, DirectoryBaseViewModel directory, TemplateAssetDescription templateDescription, IList<UFile> files, PropertyContainer? customParameters)
+        {
+            List<AssetViewModel> newAssets = new List<AssetViewModel>();
+            if (files is not null)
+            {
+                for (int i = 0; i < files.Count(); i++)
+                {
+                    var file = files[i];
+                    bool inResourceFolder = directory.Package.Package.ResourceFolders.Any(x => file.FullPath.StartsWith(x.FullPath));
+
                     if (inResourceFolder == false)
                     {
-                        var message = Tr._p("Message", "Source file '{0}' is not inside any of your project's resource folders, import anyway ?");
-                        message = string.Format(message, file.FullPath);
+                        var message = Tr._p("Message", "Source file '{0}' is not inside of your project's resource folders, do you want to copy it?").ToFormat(file.FullPath);
+
+                        var copyResult = await Dialogs.MessageBox(message, MessageBoxButton.YesNo, MessageBoxImage.Warning);
+
+                        if (copyResult == MessageBoxResult.Yes)
+                        {
+                            string finalPath = await GetAssetCopyDirectory(directory, file);
+
+                            try
+                            {
+                                Directory.CreateDirectory(Path.GetDirectoryName(finalPath));
+                                if (File.Exists(finalPath))
+                                {
+                                    message = Tr._p("Message", "The file '{0}' already exists, it will get overwritten if you continue, do you really want to proceed?").ToFormat(file.FullPath);
+
+                                    copyResult = await Dialogs.MessageBox(message, MessageBoxButton.YesNo, MessageBoxImage.Warning);
+                                    
+                                    // Abort if the user says no or closes the prompt
+                                    if (copyResult != MessageBoxResult.Yes)
+                                    {
+                                        return newAssets;
+                                    }
+                                    File.Delete(finalPath);
+                                }                                
+                                
+                                File.Copy(file.FullPath, finalPath);
```

Comment:
Maybe copy with overwrite [Copy(String, String, Boolean)](https://learn.microsoft.com/en-us/dotnet/api/system.io.file.copy?view=net-7.0#system-io-file-copy(system-string-system-string-system-boolean)) when it exists, and use normal copy if the file doesn't.
That way it's an atomic operation, if a file somehow moves there between the delete and copy it won't fail. 
Very minor though, feel free to ignore.